docs/source/form-config.rst

```diff
@@ -0,0 +1,78 @@
+Form configuration files
+========================
+
+
+Introduction
+------------
+
+Form configuration files are an implementation-independent means of specifying the content of Impact Stack forms; by using JSON syntax to specify parent-child relationships between form components and key-value relationships between data, front- and back-end routines can use the same structure for their own purposes, e.g. to generate corresponding HTML or to validate form submissions.
+
+
+Structure
+---------
+
+The structure of a form configuration file is as follows:
+
+- Form components are represented as nodes in a tree structure, where the root node typically represents the component that generates a ``<form>`` element.
+- The root node *may* have a ``context`` attribute in which to store key-value data pertaining to its use with logCRM.
+- Each node **must** have a ``type`` attribute corresponding to the filename in which the component's template is stored.
+- Each node *may* have a ``params`` attribute in which to store key-value data used by the component: there are no fixed requirements for its content, as it depends on which data the component template requires and how it uses it, but a useful convention is to match attribute/property names and values available on the component's principal HTML element where possible.
+- Each node *may* have a ``components`` attribute which contains a list of sub-components that are to be generated from within the current node's component template; the manner in which it does this is up to the template.
```

Comment:
The explanations take very much a formtool-frontend specific perspective (the word “template“ occurs 4 times on 3 points). It prefer finding a more neutral explanations for the type/attributes.

Reply:
I agree... the correspondance between how the components are represented in the config and how they're represented by the HTML templates has been my primary focus for the last month, so I'm not surprised that my write-up has been rather influenced by the experience!

I do think it's important to make that link clear, as I see it as the main premise of the abstraction, but I also see that there's nothing about how the form config is subsequently used on the back-end... I'm not familiar enough with the code you and Katharina have written to contribute good documentation on it, though, I think!

Reply:
Thanks for making the edits! It's quite interesting to observe the differences in the way we see the components: for example, I wouldn't have thought to say that the `type` attribute defines the behaviour of the component, I would have said that the contents of its template does... and while that's absolutely true looking at it from the front-end, it's much clearer from how you've expressed it that the `type` has meaning for validation behaviour on the back-end! :)